lib/forspell/reporter.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'pastel'
+require 'logger'
+require 'json'
+
+module Forspell
+  class Reporter
+    SUCCESS_CODE = 0
+    ERROR_CODE = 1
+    FORMAT = '%<file>s:%<line>i: %<text>s (suggestions: %<suggestions>s)'
+
+    def initialize(logfile:,
+                   verbose:,
+                   format:,
+                   group:)
+
+      FileUtils.touch(logfile) if logfile.is_a?(String)
+      @logger = Logger.new(logfile || STDOUT)
+      @logger.level = verbose ? Logger::INFO : Logger::WARN
+
+      @logger.formatter = proc do |_severity, _datetime, _progname, msg|
+        "#{msg}\n"
+      end
+
+      @format = group ? 'group' : format
+
+      @pastel = Pastel.new(enabled: $stdout.tty?)
+      @errors = []
+      @files = []
+    end
+
+    def file(path)
+      @logger.info "Processing #{path}"
+      @files << path
+    end
+
+    def error(word, suggestions)
+      @errors << [word, suggestions]
+      puts readable(word, suggestions) if @format == 'readable'
+    end
+
+    def parsing_error(error)
+      @logger.error "PARSING ERROR IN #{@files.last}: #{error}"
```

Comment:
Don't scream :) If you want everybody to look at it, maybe it is worth to have in log formatter colorization... Or adding level severity before the line. But just `Parsing error` is visible enough